proxy.js

```diff
@@ -9,79 +8,118 @@ const RACEACTIONKEYNAME = 'raceaction';
-        endpoint = '/racestatus';
+	var obj = JSON.parse(jsonString);
 
-        if(obj.racestatus.raceAction == 'start') {
-            raceStarted = true;
-        }
-        else if(obj.racestatus.raceAction =='abort' || obj.racestatus.raceAction == 'race finished') {
-            raceStarted = false;
-        }
-    }
+	var endpoint = null;
 
-    if (endpoint){
-        await postMessage(endpoint, JSON.stringify(obj));
-    }
+	if (obj[RACEDATAKEYNAME] != null && shouldSend(obj[RACEDATAKEYNAME])) {
+		obj = buildRaceDataForSending(obj);
+		endpoint = '/racedata';
+	}
+	else if (obj[RACESTATUSKEYNAME] != null) {
+		endpoint = '/racestatus';
+
+		if (obj.racestatus.raceAction == 'start') {
+			raceStarted = true;
+			pilotDictionary = {};
+		}
+		else if (obj.racestatus.raceAction == 'abort' || obj.racestatus.raceAction == 'race finished') {
+			raceStarted = false;
+		}
+	}
+
+	if (endpoint) {
+		await postMessage(endpoint, JSON.stringify(obj));
+	}
+}
+
+function buildRaceDataForSending(data) {
+	let returnObj = {};
+	returnObj[RACEDATAKEYNAME] = {};
+
+	for (let pilotName of Object.keys(data[RACEDATAKEYNAME])) {
+		let pilotData = data[RACEDATAKEYNAME][pilotName];
+
+		let shouldSend = shouldSendForPilot(pilotData);
+
+		if (shouldSend == true && pilotDictionaryHasPilot(pilotName) == false) {
+			pilotDictionary[pilotName] = createPilot(pilotData, pilotName);
+			returnObj[RACEDATAKEYNAME][pilotName] = data[RACEDATAKEYNAME][pilotName];
+		}
+		else if (shouldSend == true) {
+			pilotDictionary[pilotName].times.push(pilotData.time);
+			returnObj[RACEDATAKEYNAME][pilotName] = data[RACEDATAKEYNAME][pilotName];
+		}
+	}
+
+	return returnObj;
+}
+
+function shouldSend(data) {
+	if (raceStarted == false) {
+		return false;
+	}
+
+	for (let pilotName of Object.keys(data)) {
+		let pilotData = data[pilotName];
+
+		return shouldSendForPilot(pilotData, pilotName);
+	}
+}
+
+function shouldSendForPilot(pilotData, pilotName) {
+	var pilot = getPilotFromDictionary(pilotName);
+
+	return pilot == null || pilot.times[pilot.times.length - 1] != pilotData.time;
+}
+
+function getPilotFromDictionary(pilotName) {
+	return pilotDictionaryHasPilot(pilotName) == true ? pilotDictionary[pilotName] : null;
+}
+
+function pilotDictionaryHasPilot(pilotName) {
+	return pilotDictionary.hasOwnProperty(pilotName) == true;
+}
+
+function createPilot(pilotData, pilotName) {
+	return {
+		"name": pilotName,
+		"times": [pilotData.time]
+	};
 }
 
-async function postMessage (url, data) {
-   
-    try {
-        var res = await fetch('http://127.0.0.1:3000' + url, {
-            method: 'POST',
-            body: data,
-            headers: {
-                'Content-Type': 'application/json',
-                'x-api-key': '{USERAPIKEY}'
-            }
-        });
-    } catch(error) {
-        console.log(error);
-    }
+async function postMessage(url, data) {
+
+	try {
+		var res = await fetch('http://127.0.0.1:3000' + url, {
```

Comment:
should this be a setting?

Reply:
Yeh, I have done that now.
Just after putting in the PR I thougth "What's next?.. probably make some settings" :D